node/src/manager/commands/chain.rs

```diff
@@ -143,3 +152,55 @@ pub fn remove(primary: ConnectionPool, store: Arc<BlockStore>, name: String) ->
 
     Ok(())
 }
+
+pub fn change_block_cache_shard(
+    primary_store: ConnectionPool,
+    store: Arc<BlockStore>,
+    chain_name: String,
+    shard: String,
+) -> Result<(), Error> {
+    println!("Changing block cache shard for {} to {}", chain_name, shard);
+
+    let conn = primary_store.get()?;
+    let chain = find_chain(&conn, &chain_name)?.expect("no such chain");
```

Comment:
Instead of `expect`, `.ok_or_else(|| anyhow!("unknown chain: {}", name))?;` would produce a friendlier error